clarsach/spectrum.py

```diff
@@ -30,17 +30,17 @@ def __init__(self, filename, telescope='HETG'):
         if self.bin_unit != self.rmf.energ_unit:
             print("Warning: RMF units and pha file units are not the same!!!")
 
-
-        # Might need to notice or group some day
-        #self.notice = np.ones(len(self.counts), dtype=bool)
-        #self.group  = np.zeros(len(self.counts), dtype=int)
-
-
     def __store_path(self, filename):
         self.path = '/'.join(filename.split('/')[0:-1]) + "/"
-
         return
 
+    def apply_resp(self, mflux):
+        # Given a model flux spectrum, apply the response
+        mrate  = self.arf.apply_arf(mflux)  # phot/s per bin?
```

Comment:
Do we think that this should be applicable more generally than for HETG? If so, it might be worth having an `if` statement before this line to check whether there is an `ARF` object assigned to `self.arf`, and only apply it if `self.arf` is not `None`, since there are some instruments who don't have an ARF.